squared_equation.rb

```diff
@@ -0,0 +1,20 @@
+def solve_equation(a, b, c)
+  d = b**2 - 4 * a * c
+  puts "D = #{d}"
+  return nil if d.negative?
+
+  results = []
+  results.push((-b + Math.sqrt(d)) / 2 * a)
+  results.push((-b - Math.sqrt(d)) / 2 * a)
+  results.uniq
+end
+
+
+print 'Enter a: '
```

Comment:
-5x^2 - x + 14 = 0
D = 281, it is correct, but roots is 1.5 and -1.7
in your case, it is -44.4 and 39.4
2x^2-16x+0 = 0
D = 256, is correct
but roots are not!
check the logic, please